src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -29,6 +29,72 @@
 namespace Dynamo.Graph.Workspaces
 {
     /// <summary>
+    /// Non view-specific container for additional view information required to
+    /// fully construct a WorkspaceModel from JSON
+    /// </summary>
+    public class ExtraViewInfo
+    {
+        public object Cameras;
+        public IEnumerable<ExtraNodeViewInfo> NodeViews;
+        public IEnumerable<ExtraNoteViewInfo> Notes;
+        public IEnumerable<ExtraAnnotationViewInfo> Annotations;
+        public double X;
+        public double Y;
+        public double Zoom;        
+    }
+
+    /// <summary>
+    /// Non view-specific container for additional node view information 
+    /// required to fully construct a WorkspaceModel from JSON
+    /// </summary>
+    public class ExtraNodeViewInfo
+    {
+        public string Id;
+        public string Name;
+        public double X;
+        public double Y;
+        public bool IsVisible;
+        public bool IsUpstreamVisible;
+    }
+
+    /// <summary>
+    /// Non view-specific container for additional note view information 
+    /// required to fully construct a WorkspaceModel from JSON
+    /// </summary>
+    public class ExtraNoteViewInfo
+    {
+        public string Id;
+        public double X;
+        public double Y;
+        public string Text;
+
+        // TODO: Figure out if this is necessary
+        //public int ZIndex;
+    }
+
+    /// <summary>
+    /// Non view-specific container for additional annotation view information 
+    /// required to fully construct a WorkspaceModel from JSON
+    /// </summary>
+    public class ExtraAnnotationViewInfo
+    {
+      public string Title;
+      public IEnumerable<string> Nodes;
+      public double FontSize;
+      public string Background;
+
+      // TODO: Determine if these are required
+      //public string Id;
+      //public double Left;
+      //public double Top;
+      //public double Width;
+      //public double Height;
+      //public double InitialTop;
+      //public double InitialHeight;
+      //public double TextBlockHeight;
```

Comment:
They are required to display annotations, but the values are created by the view values of the nodes that are contained inside the annotations. I was not sure if it was necessary to set these values directly or indirectly, letting the values get set from the contained nodes seemed a more standard way than trying to force them to the saved values. I can set them directly if this is determined to be a better way to go.